### PR TITLE
f-enhancement/gh-1104-validate-carried-over-days-expiration-date

### PR DESCRIPTION
**Pull request by fkis1**
Newly created years will have a default value of `Expiration date for carried over days` set to 1st of April of the following year (end of first year quarter). 

Resolves: #1104

### CODE REVIEW

File: src/main/frontend/src/store/helperFunctions.js [hunk 1]
```diff
@@ -3,8 +3,10 @@ import moment from 'moment';
 
 export const reducerTotal = (a, b) => a + b;
 export const formatDate = date => moment(date).format('DD MMM YYYY');
-export const formatExpirationDate = date =>
-  !date ? date : moment(date).toDate();
+export const formatExpirationDate = (date, year) =>
```
Comment (jolpazy) [inline src/main/frontend/src/store/helperFunctions.js]: please make a formatter with meaningful name for thing like this, so we could know what it does when we see it

Comment (fkis1) [reply src/main/frontend/src/store/helperFunctions.js]: Thanks for the input. I changed it to `initializeOrFormatCarriedOverHoursExpirationDate`. I am not sure if the new name too long or not adequately descriptive.

File: src/main/frontend/src/pages/administration/components/years/EditYear.js [hunk 2]
```diff
@@ -66,15 +66,16 @@ class EditYear extends Component {
...
     if (!this.props.year) return null;
 
-    const { carriedOverHoursExpirationDate } = this.props.year;
+    const { carriedOverHoursExpirationDate, year } = this.props.year;
 
     return (
       <Formik
         validationSchema={validationSchema}
         initialValues={{
           ...this.props.year,
```
Comment (jolpazy) [inline src/main/frontend/src/pages/administration/components/years/EditYear.js]: can we generally try and initailize big variables outside JSX (initialValues)

Comment (fkis1) [reply src/main/frontend/src/pages/administration/components/years/EditYear.js]: Okay, noted! I extracted the initial values to a variable outside JSX

**Pull request merged**